src/hotspot/share/code/nmethod.hpp

```diff
@@ -200,18 +200,20 @@ class nmethod : public CodeBlob {
   static nmethod*    volatile _oops_do_mark_nmethods;
   oops_do_mark_link* volatile _oops_do_mark_link;
 
+  CompiledICData* _compiled_ic_data;
+
   // offsets for entry points
-  address _entry_point;                      // entry point with class check
-  address _verified_entry_point;             // entry point without class check
-  address _osr_entry_point;                  // entry point for on stack replacement
+  address  _osr_entry_point;       // entry point for on stack replacement
+  uint16_t _entry_offset;          // entry point with class check
+  uint16_t _verified_entry_offset; // entry point without class check
```

Comment:
Changed direct entry pointers (8 bytes each in 64-bit VM) to offsets (2 bytes) to `code_begin()`.